packages/core/src/metric-summary.ts

```diff
@@ -0,0 +1,56 @@
+import type { MeasurementUnit, Primitive, SpanMetricSummaryAggregator } from '@sentry/types';
+import type { MetricSpanSummary } from '@sentry/types';
+import { dropUndefinedKeys } from '@sentry/utils';
+
+/** */
+export class MetricSummaryAggregator implements SpanMetricSummaryAggregator {
+  private readonly _measurements: Map<string, MetricSpanSummary>;
+
+  public constructor() {
+    this._measurements = new Map<string, MetricSpanSummary>();
+  }
+
+  /** @inheritdoc */
+  public add(
+    metricType: 'c' | 'g' | 's' | 'd',
+    name: string,
+    value: number,
+    unit?: MeasurementUnit | undefined,
+    tags?: Record<string, Primitive> | undefined,
+  ): void {
+    const exportKey = `${metricType}:${name}@${unit}`;
+    const bucketKey = `${exportKey}\n${JSON.stringify(tags)}`;
+
+    const summary = this._measurements.get(bucketKey);
+
+    if (summary) {
```

Comment:
If the metric type is `set`, you set the value to `1.0`.

Reply:
Does this mean when metric type is NOT `s`, value should always be `number`? 
If so it might be worth tightening our types around the `Sentry.metric.*` methods...

Reply:
See https://github.com/getsentry/sentry-python/blob/c8e917205d77e143c68c0cda4df877167d402830/sentry_sdk/metrics.py#L545-L554C33, there is a special handling for sets, even if the value is not a string.

Reply:
Is there somewhere I can read specifically about sets so I can more fully understand why this is done?

Reply:
ah found [this](https://develop.sentry.dev/sdk/metrics/#sets) and [this](https://github.com/getsentry/rfcs/blob/main/text/0123-metrics-correlation.md#sets)